Doc for Quable (#3085)
ezsystems/developer-documentation: docs/product_catalog/quable/configure_quable_connector.md

---

> description: Quable PIM connector configuration reference for Ibexa DXP
> page_type: reference

Reviewer: ```suggestion
description: Quable PIM connector configuration reference for Ibexa DXP
page_type: reference
month_change: true
```

---

> description: Quable PIM connector configuration reference for Ibexa DXP

Reviewer 2: We should add Quable and Raptor to vale rules, like Ibexa, so that they're always coming from a variable.

Author: Done in https://github.com/ibexa/vale-styles/pull/38 !

Please note that variables cannot be used in H1 title and descriptions now anyway - something I tackle in https://github.com/ibexa/documentation-developer/pull/3116 (to be continued after the release)

page_type: reference
month_change: true
---

# Configure Quable connector

> You can customize the behavior of the [[= pim_product_name =]] integration by using the following [configuration](configuration.md).

Reviewer 2: ```suggestion
You can customize the behavior of the [[= pim_product_name =]] integration add-on by using the following [configuration](configuration.md).
```

I suggested the same to Julita, that we should treat Quable integration or Raptor integration as objects of the "add-on" noun (technical term). Just like we treat LTS updates. This way we can link to a list of add-ons, and we do not make "integration" sound like it is our technical term.

        product_type: true
```

## Configuration options

| Parameter | Default value | Description |
|-----------|--------------------------|-------------|
| `enabled` | `false` | Enables the connector. |
| `instance_url` | string | Base URL of your [[= pim_product_name =]] instance, for example `https://example.quable.com`. |
| `api_token` | string | [Read Access API token](https://docs.quable.com/v5-EN/docs/system-api-tokens) used to authenticate requests to [[= pim_product_name =]]. |
| `channel_code` | string | Code of the [[[= pim_product_name =]] channel](https://docs.quable.com/v5-EN/docs/content-channels) used as the source of product data. |
| `webhook_secret` | string | Secret expected in the [webhook](https://docs.quable.com/v5-EN/docs/webhook) authorization header. |
| <nobr>`throw_on_invalid_criteria`</nobr> | <nobr>`%kernel.debug%`</nobr> | Controls behavior for unsupported search criteria: `true` throws an exception, `false` only logs unsupported criteria. |
| <nobr>`throw_on_invalid_mapping`</nobr> | <nobr>`%kernel.debug%`</nobr> | Controls behavior for mapping errors during data transformation: `true` throws an exception, `false` only logs mapping errors. |

> | `cache.enabled` | `true` | Global cache switch for the connector. When `false`, cache decorators use only [in-memory cache](persistence_cache.md#in-memory-cache-configuration). |

Reviewer 2: ```suggestion
| `cache.enabled` | `true` | Global cache switch for the connector. When set to `false`, cache decorators use only [in-memory cache](persistence_cache.md#in-memory-cache-configuration). |
```
By analogy to other occurrences of similar phrases

> | `cache.enabled` | `true` | Global cache switch for the connector. When set to `false`, cache decorators use only [in-memory cache](persistence_cache.md#in-memory-cache-configuration). |

Reviewer 3: ```suggestion
| `cache.enabled` | `true` | Global cache switch for the connector. When set to `false`, only [in-memory cache](persistence_cache.md#in-memory-cache-configuration) is used. |
```

Also, I'd like to note that we are using `cache.app.taggable` (so indirectly `cache.app`) to create caching services, which means we are **using the cache that is declared for Symfony framework**. See https://symfony.com/doc/current/cache.html#system-cache-and-application-cache

> `cache.app` is a general-purpose data cache for application and bundle code. Data in this pool does not need to be flushed on deployment. It defaults to `cache.adapter.filesystem`, but configuring a faster adapter like Redis is recommended when available (this ensures cached data survives deployments and is shared across multiple instances in a multi-server setup).

Author: Added as:
> When set to `true`, [Symfony's `cache.app` cache pool]([[= symfony_doc =]]/cache.html#system-cache-and-application-cache) is used.

linking to https://symfony.com/doc/current/cache.html#system-cache-and-application-cache

| `cache.attribute` | `true` | Enables caching for attribute definition requests. |
| `cache.`<br>`attribute_group` | `true` | Enables caching for attribute group requests. |
| `cache.`<br>`product` | `true` | Enables caching for product requests. |
| `cache.`<br>`product_type` | `true` | Enables caching for product type requests. |

In production environments, it's recommended to:

- keep the `api_token` and the `webhook_secret` [secure](security_checklist.md#app_secret-and-other-secrets)

> - enable caching for better performance, using Redis or Valkey as [persistence cache](persistence_cache.md#redisvalkey)

Reviewer 2: ```suggestion
- enable caching for better performance, by using Redis or Valkey as [persistence cache](persistence_cache.md#redisvalkey)
```
